src/python/pants_release/generate_release_announcement.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import Path
+
+from pants_release.common import sorted_contributors
+from pants_release.git import git
+
+from pants.util.strutil import softwrap
+
+
+def get_new_contributors() -> list[str]:
+    cur_version_sha, prev_version_sha = git(
+        "log", "-2", "--pretty=format:%h", "src/python/pants/VERSION"
+    ).splitlines(keepends=False)
+    return [
+        line[3:]
+        for line in git(
+            "diff", f"{prev_version_sha}..{cur_version_sha}", "CONTRIBUTORS.md"
+        ).splitlines(keepends=False)
+        if line.startswith("++ ")
+    ]
+
+
+def announcement_text() -> str:
+    version = Path("src/python/pants/VERSION").read_text().strip()
+    cur_version_sha, prev_version_sha = git(
+        "log", "-2", "--pretty=format:%h", "src/python/pants/VERSION"
+    ).splitlines(keepends=False)
+    git_range = f"{prev_version_sha}..{cur_version_sha}"
+    all_contributors = sorted_contributors(git_range)
+    new_contributors = sorted(
+        [
+            line[3:]
+            for line in git("diff", git_range, "CONTRIBUTORS.md").splitlines(keepends=False)
+            if line.startswith("++ ")
+        ]
+    )
+
+    announcement = softwrap(
```

Comment:
Would it make sense to start using Jinja for string templates? I find string templates much more readable and easier to debug.

Reply:
I don't think so for this change. It's too trivial. And we're trying to reduce deps in the repo, right?

Reply:
For a more substantial thing, possibly? But then we'd have to take steps to ensure it doesn't end up used in the pants binary itself? So not sure it's worth the complication.

Reply:
We could do that with separate resolves I suppose! But anyway, for a followup at best. And I think I'd like to see a more acute need?

Reply:
We can and should use 3rdparty for our helper scripts. Again, I love 3rdparty :stuck_out_tongue: I just think build tools are themselves are special :smile:

Reply:
I also don't think this so so trivial it wouldn't need a template. all the strings and newlines and branching and loops makes my brain just turn off :joy: 

But that could just be me.

Reply:
> We can and should use 3rdparty for our helper scripts. Again, I love 3rdparty 😛 I just think build tools are themselves are special 😄

But then how do you prevent people from using it in Pants itself? Now we need more guardrails.

Reply:
I like avoiding any 3rd party unless justified. Fwiw on guards, we could use the visibility rules introduced to ensure anything under SRC/pants only has access to a specific set of dependencies

Reply:
Yeah we're currently using a sophisticated build tool with guardrails built in :wink: 

But also, we have "guardrails" today, [they're just made of paper-mache](https://github.com/pantsbuild/pants/blob/b6ec890797088bc974404125ffcf2a580f0be99a/3rdparty/python/requirements.txt#L1)